DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/src/svd.hpp

```diff
@@ -0,0 +1,273 @@
+#ifndef __SVD_HPP__
+#define __SVD_HPP__
+
+#include <sycl/ext/intel/ac_types/ac_complex.hpp>
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+#include <vector>
+
+#include "exception_handler.hpp"
+#include "memory_transfers.hpp"
+#include "non_std_convariance.hpp"
+#include "post_process.hpp"
+#include "streaming_eigen.hpp"
+#include "streaming_qrd.hpp"
+#include "svd_helper.hpp"
+#include "tuple.hpp"
+
+// kernels
+class IDInputMatrixFromDDRToLocalMem;
+class IDInputTransMatrixFromDDRToLocalMem;
+class IDCovarianceMatrixComputation;
+class IDEigenValuesAndVectorsComputation;
+class IDInputMatrixFromDDRToLocalMemPostProcess;
+class IDUSVFromEigens;
+class IDOrthogonalizingU;
+class IDRankDeficientFlagFromLocalMemToDDR;
+class IDUMatrixFromLocalMemToDDR;
+class IDRMatrixFromLocalMemToDDR;
+class IDSMatrixFromLocalMemToDDR;
+class IDVMatrixFromLocalMemToDDR;
+
+// pipes
```

Comment:
Consider expanding these to match the pipe names in your diagram. Expanding the names will make the reports easier to read.

Reply:
Trying to see if we have any convention on pipe id (like 'ID' prefix for kernel id) in code sample, but couldn't find any. I will just use ID, just so its not the same as the variable name.